src/platform/nxp/mw320/KeyValueStoreManagerImpl.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Platform-specific key value storage implementation for MW320
+ */
+/* this file behaves like a config.h, comes first */
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#include "lib/support/CHIPMem.h"
+#include <platform/KeyValueStoreManager.h>
+#include <platform/nxp/mw320/MW320Config.h>
+#include <string>
+extern "C" {
+#include <network_flash_storage.h>
+}
+
+
+namespace chip {
+namespace DeviceLayer {
+namespace PersistedStorage {
+
+/* TODO: adjust this value */
+#define MAX_NO_OF_KEYS 255
+
+KeyValueStoreManagerImpl KeyValueStoreManagerImpl::sInstance;
+
+#define kMaxKeyValueBytes 2048
+CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t value_size, size_t * read_bytes_size,
+                                          size_t offset_bytes)
+{
+    uint32_t ret;
+    uint32_t read_size=kMaxKeyValueBytes;
+    uint8_t buf[kMaxKeyValueBytes];
+
+    VerifyOrReturnError(value != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Should be allowed if value_size is 0, no?

Reply:
To be clear: value being nullptr is _allowed_ as long a value_size is 0.  Which is also allowed.  That's how consumers ask "how big is the value?"

Reply:
Please help to review commit c5cd5e1b76aa175f38ab0352fab6c1b6d25cb5f4.